### ISSUE TITLE
Access control defaults incorrect for Netdata served over a UNIX socket.

### ISSUE THREAD

**Issue by Ferroin** [netdata/netdata#8962]
##### Bug report summary
Per the [documentation](https://learn.netdata.cloud/docs/agent/web/server/#binding-netdata-to-multiple-ports), no access controls should be applied by default to connections over a UNIX socket. However, at least some calls to the agent over a UNIX socket re blocked as if access controls were in place.

##### OS / Environment

```
NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo/Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
```

##### Netdata version
v1.22.0
 

##### Component Name
area/web

##### Steps To Reproduce

1. Configure Netdata to listen on a UNIX socket.
2. Attempt to fetch `netdata.conf` over the UNIX socket.

##### Expected behavior

Netdata should behave consistently with the documentation.

**Comment by cakrit** [comment]
Which type of access controls actually blocked access? Perhaps it's the doc that needs to be updated...

**Comment by Ferroin** [comment]
Don't know how this slipped off my radar.

Currently, the only endpoint I know of which is blocked b default when no ACLs are specified is `/netdata.conf`.

**Comment by Ferroin** [comment]
And it seems that it might not be just UNIX sockets, see #9765

**Comment by cakrit** [comment]
#9765 sounds like the way it's supposed to work, I expect the user isn't trying to access netdata.conf from a local IP and would need to change their ACL. 
@thiagoftsm or @amoss who can look into the code to see what's happening with UNIX sockets? In #5181 I had added a "temporary" (yeah right), fix to allow all requests coming via a socket. See e.g. 
https://github.com/netdata/netdata/blob/master/libnetdata/socket/socket.c#L386

**Comment by amoss** [comment]
That might get overwritte, I had a quick look and the problem will be:
```
static-threaded.c:28   web_client_initialize_connection(w);
...
web_server.c:48        web_client_update_acl_matches(w);
...
web_server.c:121 
    if (!web_allow_netdataconf_from ||
       connection_allowed(w->ifd, w->client_ip, w->client_host, sizeof(w->client_host),
                          web_allow_netdataconf_from, "netdata.conf", web_allow_netdataconf_dns))
        w->acl |= WEB_CLIENT_ACL_NETDATACONF;
```

The assumption in that code is that the ip addresses are valid. There is a simple pattern in the config default `fd*` to cover the unix socket case but the `client_ip` might not be set to an appropriate filename.

**Comment by thiagoftsm** [comment]
We were missing the `UNKNOWN` value inside `web_allow_netdataconf_dns`, and thanks this we could not access `netdata.conf` when we were using the `kickstart-static64`. 

I would like to call attention for the fact that this problem was happening, if and only if, we were using `kickstart-static64` to install Netdata, I tested different nginx configurations when I did the normal installation and I never had problems to use unix sockets to access `netdata.conf`. This was not the first time that I saw a problem happening only when Netdata is installed using `kickstart-static64`.
